h2o-core/src/main/java/water/persist/PersistHex.java

```diff
@@ -0,0 +1,100 @@
+package water.persist;
+
+import water.DKV;
+import water.H2O;
+import water.Key;
+import water.Value;
+import water.fvec.C1NChunk;
+import water.util.FrameUtils;
+import water.util.Log;
+
+import java.io.*;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PersistHex extends Persist {
+
+  static final String HEX_PATH_PREFIX = PersistManager.Schemes.HEX + "://";
+
+  Key<?> fromHexPath(String path) {
+    Key<?> key = Key.make(path.substring(HEX_PATH_PREFIX.length()));
+    if (! key.isChunkKey()) {
+      throw new IllegalArgumentException("Only Chunk keys are supported for HEX schema");
+    }
+    return key;
+  }
+
+  InputStream open(Key<?> key) {
+    byte[] bytes = ((C1NChunk) DKV.getGet(key)).getBytes();
```

Comment:
Knit: `Objects.requireNonNull(key);` would be a nice addition in here, as the unlikely NPE would be clear.

Reply:
thank you, fixed